docs/modules/cloud/device_controller.md

```diff
@@ -0,0 +1,68 @@
+## Operations Performed By Device Controller
+ 
+ The following are the functions performed by the device controller :-
+ - **Downstream Controller**: Synchronize the device updates from the cloud to the edge node, by watching on K8S API server
+ - **Upstream Controller**: Synchronize the device updates from the edge node to the cloud using device twin component
+
+
+## Upstream Controller:
+
+The upstream controller watches for updates from the edge node and applies these updates against the API server in the cloud. Updates are categorized below along with the possible
+actions that the upstream controller can take:
+
+  | Update Type                        | Action                                        |
+  |-------------------------------     |---------------------------------------------- |
+  |Device Twin Reported State Updated    |  The controller patches the reported state of the device twin property in the cloud. |
+  
+
+<img src="../../images/device-crd/device-upstream-controller.png">
+     
+### Syncing Reported Device Twin Property Update From Edge To Cloud
+
+The mapper watches devices for updates and reports them to the event bus via the MQTT broker. The event bus sends the reported state of the device to the device twin which stores it locally and then syncs the updates to the cloud. The device controller watches for device updates from the edge ( via the cloudhub ) and updates the reported state in the cloud.
+
+<img src="../../images/device-crd/device-updates-edge-cloud.png">
+
+
+## Downstream Controller:
+
+The downstream controller watches for device updates against the K8S API server. Updates are categorized below along with the possible actions that the downstream controller can take:
+
+| Update Type                    | Action                                       |
+|-------------------------------|---------------------------------------------- |
+|New Device Model Created       |  NA                                           |
+|New Device Created             | The controller creates a new config map to store the device properties and visitors defined in the device model associated with the device.  This config map is stored in etcd. The existing config map sync mechanism in the edge controller is used to sync the config map to the egde. The mapper application running in a container can get the updated config map and use the property and visitor metadata to access the device. The device controller additionally reports the device twin metadata updates to the edge node.|
+|Device Node Membership Updated | The device controller sends a membership update event to the edge node.|
+|Device  Twin Desired State Updated | The device controller sends a twin update event to the edge node.|
+|Device Model Updated           |  TODO: What happens to existing devices using this model which are generating telemetry data as per the old model ? Do we update the config map associated with the devices which are using this device model ?|
```

Comment:
remove this row as it is not clear. in the docs , we publish only what is available.

Reply:
Done, thanks